src/Primer.ts

```diff
@@ -1,252 +1,409 @@
+  handleFailure: async (errorMessage: string) => {
+    try {
+      RNPrimer.handleErrorMessage(errorMessage || "");
+    } catch (err) {
+      console.error(err);
+    }
+  }
 }
 
-export const PrimerNativeMapping: IPrimer = {
+let primerSettings: PrimerSettings | undefined = undefined;
 
-  showUniversalCheckout(clientToken: string, config: IPrimerConfig): void {
-    RNPrimer.removeAllListeners();
+export const Primer: IPrimer = {
 
-    if (config.settings || config.theme) {
-      RNPrimer.configure(config.settings || null, config.theme || null);
+  ///////////////////////////////////////////
+  // SDK API
+  ///////////////////////////////////////////
 
-      let implementedRNCallbacks: any = {
-        // isClientTokenCallbackImplemented: (config.onClientTokenCallback !== undefined),
-        isTokenAddedToVaultImplemented: (config.onTokenAddedToVault !== undefined),
-        isOnResumeSuccessImplemented: (config.onResumeSuccess !== undefined),
-        isOnResumeErrorImplemented: (config.onError !== undefined),
-        isOnCheckoutDismissedImplemented: (config.onDismiss !== undefined),
-        isCheckoutFailedImplemented: (config.onError !== undefined),
-        isClientSessionActionsImplemented: (config.onClientSessionActions !== undefined)
-      };
+  async configure(settings?: PrimerSettings): Promise<void> {
+    primerSettings = settings;
+    return new Promise(async (resolve, reject) => {
+      try {
+        await RNPrimer.configure(settings);
+        resolve();
+      } catch (err) {
+        reject(err);
+      }
+    });
+  },
 
-      RNPrimer.setImplementedRNCallbacks(implementedRNCallbacks)
-        .then(() => {
-          RNPrimer.addListener('onClientSessionActions', data => {
-            const clientSessionActions: IClientSessionAction[] = data;
-            if (config.onClientSessionActions) {
-              config.onClientSessionActions(clientSessionActions, resumeHandler);
+  async showUniversalCheckout(clientToken: string): Promise<void> {
+    return new Promise(async (resolve, reject) => {
+      try {
+        RNPrimer.removeAllListeners();
+
+        let implementedRNCallbacks: any = {
+          onCheckoutComplete: (primerSettings?.onCheckoutComplete !== undefined),
+          onBeforePaymentCreate: (primerSettings?.onBeforePaymentCreate !== undefined),
+          onBeforeClientSessionUpdate: (primerSettings?.onBeforeClientSessionUpdate !== undefined),
+          onClientSessionUpdate: (primerSettings?.onClientSessionUpdate !== undefined),
+          onTokenizeSuccess: (primerSettings?.onTokenizeSuccess !== undefined),
+          onResumeSuccess: (primerSettings?.onResumeSuccess !== undefined),
+          onDismiss: (primerSettings?.onDismiss !== undefined),
+          onCheckoutFail: (primerSettings?.onCheckoutFail !== undefined),
+        };
+
+        await RNPrimer.setImplementedRNCallbacks(implementedRNCallbacks);
+
+        if (implementedRNCallbacks.primerDidCompleteCheckoutWithData) {
```

Comment:
All of these will fail, as they are checking for wrong callbacks, let's fix that @EvansPie